src/segtree.rs

```diff
@@ -107,6 +149,10 @@ impl<M: Monoid> Segtree<M> {
         self.d[p + self.size].clone()
     }
 
+    pub fn get_slice(&self) -> &[M::S] {
+        &self.d[self.size..(self.size + self.n)]
```

Comment:
Just showing off a trick: did you know that you can write like this 😜?  (I prefer this style becuase you need to specify the same variable twice.)

```suggestion
        &self.d[self.size..][..self.n]
```